### PR TITLE
Feature_738/Add callback url functionality to converters

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Issue: https://github.com/unfoldingWord-dev/door43.org/issues/738

ConvertHandler - added support for callback

### CODE REVIEW

File: libraries/lambda_handlers/convert_handler.py [hunk 1]
```diff
@@ -36,4 +42,22 @@ def _handle(self, event, context):
         converter = self.converter_class(source=source, resource=resource, cdn_file=cdn_file, options=options)
         results = converter.run()
         converter.close()  # do cleanup after run
+        if callback is not None:
```
Comment (richmahn) [inline libraries/lambda_handlers/convert_handler.py]: We should continue to try to keep any logic outside of the handler function as this is simply to be here to interface between the API Gateway and the actual tX code, just knowing how to deal with the event object to get tX the information it needs. I would think the callback would be actually part of the converter, not the handler to start the converter. This also helps in the tests as for all lambda_hanlder tests, it should simply mock the tX function ran (return None or whatever is returned by the API Gateway) and not test any other functionality.

So suggesting al l callback code be moved to converter.py so that if we ever do move off of Lambda, we don't lose this code.

File: libraries/lambda_handlers/convert_handler.py [hunk 2]
```diff
@@ -36,4 +42,22 @@ def _handle(self, event, context):
...
+            self.callback_payload = json.loads(json.dumps(self.data))  # clone so we can modify
+            self.callback_payload['results'] = results  # add results to payload and call back
+            self.do_callback(callback, self.callback_payload)
         return results
+
+    def do_callback(self, url, payload):
```
Comment (richmahn) [inline libraries/lambda_handlers/convert_handler.py]: Move to converter.py like we do a callback in manager.py

File: libraries/lambda_handlers/convert_handler.py [hunk 3]
```diff
@@ -36,4 +42,22 @@ def _handle(self, event, context):
         converter = self.converter_class(source=source, resource=resource, cdn_file=cdn_file, options=options)
         results = converter.run()
         converter.close()  # do cleanup after run
+        if callback is not None:
+            self.callback_payload = json.loads(json.dumps(self.data))  # clone so we can modify
```
Comment (richmahn) [inline libraries/lambda_handlers/convert_handler.py]: I think deepcopy() is a better way to clone. See my data_utils.py for how I do it to mask sensitive fields that we log.

File: tests/lambda_handlers_tests/test_convertHandler.py [hunk 4]
```diff
@@ -1,12 +1,49 @@
...
 import mock
 from unittest import TestCase
+from requests import Response
 from libraries.lambda_handlers.convert_handler import ConvertHandler
 from libraries.converters.md2html_converter import Md2HtmlConverter
 from libraries.converters.usfm2html_converter import Usfm2HtmlConverter
 
 
 class TestConvertHandler(TestCase):
+
```
Comment (richmahn) [inline tests/lambda_handlers_tests/test_convertHandler.py]: Move to test_converter.py so we aren't testing the lambda handler function once the code is moved to converter.py.

**Review by richmahn — CHANGES_REQUESTED** [review]
Need to not be doing any logic in the handler layer for Lambda/API Gateway, as this was to be kept as simple as possible, especially now that this will be done away with if we move to flask apps, etc. Should all be contained in converter.py.

File: libraries/lambda_handlers/convert_handler.py [hunk 5]
```diff
@@ -36,4 +42,22 @@ def _handle(self, event, context):
...
+            self.callback_payload['results'] = results  # add results to payload and call back
```
Comment (richmahn) [inline libraries/lambda_handlers/convert_handler.py]: It's ok to just send back `job_id` and `results`... it doesn't need the whole job payload.

**Review by richmahn — APPROVED** [review]
LGTM
